crypto/evp/pmeth_gn.c

```diff
@@ -165,6 +167,17 @@ int EVP_PKEY_gen(EVP_PKEY_CTX *ctx, EVP_PKEY **ppkey)
     if (ctx->op.keymgmt.genctx == NULL)
         goto legacy;
 
+    /*
+     * Asssigning gentmp to ctx->keygen_info is something our legacy
+     * implementations do.  Because the provider implementations aren't
+     * allowed to reach into our EVP_PKEY_CTX, we need to provide similar
+     * space for backward compatibility.  It's ok that we attach a local
+     * variable, as it should only be useful in the calls down from here.
+     * This is cleared as soon as it isn't useful any more, i.e. directly
+     * after the evp_keymgmt_util_gen() call.
+     */
+    ctx->keygen_info = gentmp;
```

Comment:
Maybe also add `ctx->keygen_info_count = 2` ?

Reply:
Ah, right, forgot about that!

Reply:
Done